src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -91,6 +91,7 @@ internal void WriteAsPax(Stream archiveStream, Span<byte> buffer)
             // Fill the current header's dict
             CollectExtendedAttributesFromStandardFieldsIfNeeded();
             // And pass them to the extended attributes header for writing
+            _extendedAttributes ??= new Dictionary<string, string>();
             extendedAttributesHeader.WriteAsPaxExtendedAttributes(archiveStream, buffer, _extendedAttributes, isGea: false);
```

Comment:
There seem to be a lot of places where we're newing up empty dictionaries.  It'd be nice if we could leave them as null until either (a) someone asks for them (make the new empty dictionary and remember that), or (b) we need to write something into it (such as when converting), and just let the writing path handle null.

Reply:
I simplified the newing in https://github.com/dotnet/runtime/pull/70325/commits/86a9297b8d0d805467941569ec9030cbd2aaa15c